functions/lib/sendgrid/delete_event.py

```
@@ -0,0 +1,167 @@
+            "event_date": event_date,
+        }.items()
+        if value is None
+    ]
+
+    if missing_fields:
+        logger.warning(
+            f"Missing event details for eventId={request_data.eventId}. Missing fields: {', '.join(missing_fields)}."
+        )
+        return {"status": 400, "message": f"Missing event details: {', '.join(missing_fields)}"}
+
+    if event_status == False:
+        logger.info(f"Event {event_name} was already inactive at deletion. No email will be sent.")
+        return {"status": 200, "message": "Event already inactive"}
+
+    # Convert price to dollars
+    event_price = cents_to_dollars(event_price)
+    logger.info(f"Converted event price to dollars: {event_price}")
+
+    sg = SendGridAPIClient(SENDGRID_API_KEY)
+
+    # Prepare attendees list for the email template
+    attendees = [
+        {
+            "name": name,
+            "email": purchaser_info.get("email"),
+            "tickets": purchaser_info.get("totalTicketCount", 0)
+        }
+        for purchaser_info in purchaser_map.values()
+        for name, attendee_info in purchaser_info.get("attendees", {}).items()
+    ]
+
+    # Send organizer email
+    try:
+        organiser_message = Mail(
+            from_email="team.sportshub@gmail.com",
+            to_emails=organiser_email,
+            subject=f"Your event '{event_name}' has been deleted",
+        )
+        organiser_message.template_id = DELETE_EVENT_ORGANISER_EMAIL_TEMPLATE_ID
+        organiser_message.dynamic_template_data={
+            "organiser_name":"",
+            "event_name": event_name,
+            "event_date": date_string,
+            "attendees": attendees
+        }
+        sg.send(organiser_message)
+        logger.info(f"Organizer email sent to {organiser_email} for event: {event_name}")
+    except Exception as e:
+        logger.error(f"Failed to send email to organizer. Exception: {e}")
+        return {"status": 500, "message": "Failed to send organizer email"}
```

i think can we not fail fast here? we still want to send out purchaser emails, even if the organiser one fails